subsys/app_memory/app_memdomain.c

```diff
@@ -92,6 +92,19 @@ void app_calc_size(void)
  */
 void appmem_init_app_memory(void)
 {
+#ifdef CONFIG_ARC
+	/*
+	 * appmem_init_app_memory will access all partitions
+	 * For CONFIG_ARC_MPU_VER == 3, these partitions are not added
+	 * into MPU now, so need to disable mpu first to do app_bss_zero()
+	 */
+	arc_core_mpu_disable();
```

Comment:
@ruuddw @vonhust  I am confused here. This code is supposed to run in supervisor mode. But still I see that mpu needs to be disabled to access these memory regions. I don't understand why this needs to be here.
Can you please help me in understanding this.

Reply:
might need to change the perms on the memory regions so that supervisor mode has full access, what's the ARC memory domain map flags look like?

Reply:
Good suggestion to move the 'disable' code from the test application to the generic subsys code. Indeed ARC initial memory map prohibits access to the appmem addresses, it's only enabled on switching to usermode, I believe. So for this bss initialization in between initial MPU setup and switching to userspace, MPU protection is disabled.

Reply:
>Good suggestion to move the 'disable' code from the test application to the generic subsys code. Indeed ARC initial memory map prohibits access to the appmem addresses, it's only enabled on switching to usermode, I believe. So for this bss initialization in between initial MPU setup and switching to userspace, MPU protection is disabled.

Could we add some of this text in the comment, to make it more clear why we need to disable and enable?

Reply:
>Good suggestion to move the 'disable' code from the test application to the generic subsys code. Indeed ARC initial memory map prohibits access to the appmem addresses, it's only enabled on switching to usermode, I believe. So for this bss initialization in between initial MPU setup and switching to userspace, MPU protection is disabled

oh okay. I will add this in the code as comments. Will help later on.